spatialdata/_core/_spatialdata.py

```diff
@@ -1198,6 +1198,53 @@ def _gen_elements(self) -> Generator[tuple[str, str, SpatialElement], None, None
             for k, v in d.items():
                 yield element_type, k, v
 
+    def __getitem__(self, item: str) -> SpatialElement | AnnData:
+        """
+        Return the element with the given name.
+
+        Parameters
+        ----------
+        item
+            The name of the element to return.
+        Returns
+        -------
+        The element.
+        """
+        found = []
+        for _, element_name, element in self._gen_elements():
+            if element_name == item:
+                found.append(element)
+        assert len(found) <= 1
+        if len(found) == 1:
+            return found[0]
```

Comment:
```suggestion
        for _, element_name, element in self._gen_elements():
            if element_name == item:
                return element
```

Reply:
we have check for uniqueness, no need to get list and check length.

Reply:
as of now, the user can always do this:

```
sdata.images['my_element'] = image
sdata.labels['my_element'] = label
```

this would not raise any error, but would make the `sdata` object inconsistent. I agree to remove the check for uniqueness here as you suggested. But we should think how to avoid the user to reuse the same name across elements.

Possibility 1:
make `.images`, `.labels` etc private and make the user call
```
sdata['my_element'] = image
```
the correct element type will be inferred by `get_schema()` called on `image`.

Possibility 2:
we turn `.images`, `.labels` private and we replace them with classes that control the `__set_item__` and `__get_item__` behavior.c

I'll open a issue on this so we don't have to worry now.

Reply:
Issue opened here: https://github.com/scverse/spatialdata/issues/186